### PR TITLE
change way of checking image changes in pod

### PR DESCRIPTION
**Pull request by anoipm**

### CODE REVIEW

File: internal/controllers/pod_controller.go [hunk 1]
```diff
@@ -164,11 +164,31 @@ func (r *PodReconciler) labelPod(ctx context.Context, pod corev1.Pod, result val
 	return nil
 }
 
-// TODO: This function should check images not the whole spec.
-func (r *PodReconciler) areImagesChanged(oldObject runtime.Object, newObject runtime.Object) bool {
-	oldPod := oldObject.(*corev1.Pod)
-	newPod := newObject.(*corev1.Pod)
-	return !reflect.DeepEqual(oldPod.Spec.InitContainers, newPod.Spec.InitContainers) || !reflect.DeepEqual(oldPod.Spec.Containers, newPod.Spec.Containers)
+func areImagesChanged(oldObject runtime.Object, newObject runtime.Object) bool {
+	oldImages := getPodImages(oldObject.(*corev1.Pod))
```
Comment (dbadura) [inline internal/controllers/pod_controller.go:168]: Should we have `pods` as parameters?

File: internal/controllers/pod_controller.go [hunk 2]
```diff
@@ -164,11 +164,31 @@ func (r *PodReconciler) labelPod(ctx context.Context, pod corev1.Pod, result val
...
+	newImages := getPodImages(newObject.(*corev1.Pod))
+	if len(oldImages) != len(newImages) {
+		return true
+	}
+	sort.Strings(oldImages)
+	sort.Strings(newImages)
```
Comment (dbadura) [inline internal/controllers/pod_controller.go:174]: What if we have additional image in new pod?

Comment (anoipm) [reply internal/controllers/pod_controller.go:174]: IMO all is ok. Maybe I don't understand this question.

File: internal/controllers/pod_controller.go (hunk 1 above)
Comment (anoipm) [reply internal/controllers/pod_controller.go]: yes, done

File: internal/controllers/controller_test.go [hunk 3]
```diff
@@ -109,3 +109,136 @@ func Test_PodReconcile(t *testing.T) {
...
+		newPod *corev1.Pod
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "the same images in different order",
+			args: args{
```
Comment (dbadura) [inline internal/controllers/controller_test.go:125]: Can we shorten test cases to something like:
```
newImages: {
initContainers: {}
Containers:{}
}
```
and create those pods in test run?

**Review by dbadura — APPROVED** [review]

**Pull request merged**